src/xtgeo/grid3d/_grid_import_ecl.py

```diff
@@ -1,21 +1,27 @@
 """Grid import functions for Eclipse, new approach (i.e. version 2)."""
+from __future__ import annotations
 
+from typing import TYPE_CHECKING, Any, Literal
 
-import xtgeo
 from xtgeo.common import null_logger
+from xtgeo.common.sys import _XTGeoFile
 from xtgeo.grid3d._egrid import EGrid, RockModel
 from xtgeo.grid3d._grdecl_grid import GrdeclGrid, GridRelative
+from xtgeo.grid3d.grid_properties import GridProperties, gridproperties_from_file
 
 logger = null_logger(__name__)
 
 
-def import_ecl_egrid(
-    gfile,
-    relative_to=GridRelative.MAP,
-    fileformat="egrid",
-):
-    egrid = EGrid.from_file(gfile._file, fileformat=fileformat)
+if TYPE_CHECKING:
+    from xtgeo.grid3d.grid import Grid
+
 
+def import_ecl_egrid(
+    gfile: _XTGeoFile,
+    relative_to: GridRelative = GridRelative.MAP,
+    fileformat: Literal["egrid", "fegrid"] = "egrid",
+) -> dict[str, Any]:
```

Comment:
I made #1119 so that this bad-practice Any-dict pattern can be addressed in a more systematic way